pkg/sfu/streamallocator/streamallocator.go

```diff
@@ -180,14 +180,17 @@ func NewStreamAllocator(params StreamAllocatorParams) *StreamAllocator {
 		}),
 		rateMonitor: NewRateMonitor(),
 		videoTracks: make(map[livekit.TrackID]*Track),
-		eventCh:     make(chan Event, 1000),
+		eventsQueue: utils.NewOpsQueue("stream-allocator", 64),
 	}
 
 	s.probeController = NewProbeController(ProbeControllerParams{
 		Config: s.params.Config.ProbeConfig,
 		Prober: s.prober,
 		Logger: params.Logger,
 	})
+	s.eventsQueue.SetFinalize(func() {
+		s.probeController.StopProbe()
+	})
```

Comment:
this callback pattern defeats gopls call tracing and it's the single biggest obstacle to reasoning about our code

Reply:
Is there a better way to do this?

Reply:
a `<- chan struct{}` returned from the stop function would allow callers to block until the queue stops. then all the cleanup code can be written in one spot

Reply:
Sounds good. Will change it. Thank you.

Will notify once I have done that for a re-review.

Reply:
@paulwe Addressed in this [commit](https://github.com/livekit/livekit/pull/2418/commits/5e95bd18326004e6fab585427b73a6cc1bf4ddeb). Can you please double check that I have it done correctly?